rllib/core/torch/torch_sarl_trainer.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+import torch
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+from ray.rllib.core.sarl_trainer import SARLTrainer
+
+from ray.util.queue import Queue
+
+import ray.train as train
+from ray.air import session
+from ray.air.config import ScalingConfig
+from ray.train._internal.utils import construct_train_func
+from ray.train._internal.backend_executor import BackendExecutor
+from ray.train.torch import TorchConfig
+from ray.train import TrainingIterator
+from ray.train._internal.checkpoint import CheckpointManager
+from ray.train._internal.dataset_spec import RayDatasetSpec
+
+
+class TorchSARLTrainer(SARLTrainer):
+    def __init__(self, config):
+        num_workers = config.get("num_gpus", 0) or 1
+        use_gpu = bool(config.get("num_gpus", 0))
+        init_rl_module_fn = config.get("rl_module_init_fn", None)
+        module_config = config.get("rl_module_config", {})
+        """
+        we can't use fractional gpus with this :(
+
+        Leads to nccl errors:
+
+        ncclInvalidUsage: This usually reflects invalid usage of NCCL library
+        (such as too many async ops, too many collectives at once, mixing
+        streams in a group, etc).
+        """
+        scaling_config = ScalingConfig(
```

Comment:
none of these functions are framework specific yet. So they can be moved to the parent class. This will solve the confusion on where queues and training_iterator come from and what their behavior should be depends on what `self._make_ray_train_trainer` returns.